### ISSUE TITLE
Chat export 2000 MiB limit too low for attachments

### ISSUE THREAD

**Issue by dali99** [vector-im/element-web#20445]
#### What would you like to do?
I would like to export my complete chat history but currently there's a limit of 2 000 MiB
This seems kind of like an arbitrary limit, though I understand that getting all the files and zipping them on the client side could get problematic for very big exports.
Unfortunately a 1:1 chat that is 4 years old can easilly have a larger footprint like thsi

#### How would you like to achieve it?
Allow more than 2 000 MiB of data to be exported

### Have you considered any alternatives?

An alternative would be to allow paginating the export, I don't mind getting part1 part2 part3 zips each 2 GB in size.

Or if we could select date ranges like https://github.com/vector-im/element-web/issues/19310 I could manually try split it up (though that would be a pain)

**Comment by gsouquet** [comment]
All the chat exporting is done client side and everything gets loaded in RAM if I remember correctly. The max file size limit is most likely to work around that.

Taking advantage of streams could be very useful to waive this limit, but browser support was too sporadic at the time of writing this functionality

**Comment by soda-pop-ice-cream** [comment]
I think devs added 2GB restriction to protect matrix.org from huge exports.
Also, I managed to export mine 4,5GB chat before restriction was added. At first export was failed because laptop had only 8GB RAM, but then i tried on other one, that had 16GB and that was barely enough. If you don't have that much RAM, at least use swap.
@dali99 if you really need that export you can install element-web docker v1.9.1 on some server, there was no 2GB size limit.

**Comment by t3chguy** [comment]
> I think devs added 2GB restriction to protect matrix.org from huge exports.

Nope, it was added as anything larger often gets killed by the memory protections of modern browsers, even if you have far more ram the per-tab limit is far lower.

**Comment by Agent-Rogue** [comment]
for browsers, it's understandable since they wouldn't allow someone to export more than that anyway. however the native version (assuming electron doesn't have this limitation) shouldn't have this limitation.

**Comment by Szybet** [comment]
classic Matrix, such a stupid issue and still not solved after 2 years

**Comment by t3chguy** [comment]
@Szybet contributions welcome, the limit was imposed by the original external contributor of the feature based on experimental testing of common browser limitations which yield crashes

**Comment by Szybet** [comment]
Is it server side limited or client, because if server then I have tied hands

**Comment by t3chguy** [comment]
Client, there's no server-side export API, it just uses standard pagination APIs. It has to load the entire download into memory, hence the problem. Server can't help given end-to-end encryption.

**Comment by Szybet** [comment]
How do I build element desktop with a custom matrix-react-sdk

**Comment by t3chguy** [comment]
You build element-web with your matrix-react-sdk `yarn link`'ed in as per the README, then you symlink the resultant webapp into `element-desktop/webapp` and run `yarn asar-webapp` then build Element Desktop as normal ignoring the `fetch` instructions

**Comment by xmrhorizontal** [comment]
I have been waiting two years for a solution to this export problem! Developer, was it really that hard to add a 2000 MB warning ignore button!? Most importantly, element is the only client that supports exporting encrypted messages at all. I have a working chat with very important data and attachments, it was conducted for 2 years, what should I do!? How can I export it by any means!? I downloaded all versions as recommended by element web docker, but nothing works, in 1.8 there is no export at all, and in 1.9.1 pre release 1 there is already a 2000 limit. Please suggest any solution. Even the js in the browser is ready to edit, but I don't know how. Thanks!

**Comment by eastoncrafter** [comment]
Feels like this is a bigger issue, is there any progress other than running the specific element version without this restriction? I'd prefer to not have to setup a swap space and the docker container every time I need to update an export

**Comment by thegpfury** [comment]
What are the options for more than 2 GB exports? 

I have my own matrix server that I'm using for aggregation purposes, and it'd be nice to be able to export entire chats

**Comment by inducer** [comment]
#30428 is a proposed change that makes this limit a configurable option.

**Comment by inducer** [comment]
#30428 turned out to be not quite the right approach (see discussion there for details).

#30441 is a proposed change that makes at least rudimentary range export possible.
